cmd/nri-kubernetes/main.go

```diff
@@ -120,9 +122,19 @@ func main() {
 		os.Exit(exitClients)
 	}
 
+	var nsFilter *discovery.NamespaceFilter
```

Comment:
I think that no-op filter would make this PR and #457 independent. This could be merged even before, as long as, after both PRs are merged we create a follow-up PR just modifying the instance creation.

Reply:
The NOOPFilter was a great idea but doesn't make sense because the  IntegrationPopulator function in populate recieves a config but doesn't have constructor, so it can be nil, for example in the unit tests.

Reply:
Then wouldn't it make sense to include a config builder (or constructor)? We could use it both in unit tests (the `testConfig` would be replaced there) and in its usage in `scrape_jobs`.

Another alternative would to not to include the filterer in config at all. This could be achieved with a second parameter for `IntegrationPopulator` or making it a struct (now it has more responsibilities than it previously had).

Reply:
As we discussed in our call, we will leave it as it is to not make a bigger refactor (that can be done in the future in the Populator and other parts), right now the IntegrationPopulator get Namespace would be called if the filterer is not null and having the Noop doesn't add any value.